samplers/parser.go

```diff
@@ -43,6 +43,9 @@ type MetricKey struct {
 	JoinedTags string `json:"tagstring"` // tags in deterministic order, joined with commas
 }
 
+// A reusable protobuf buffer, is not safe to use concurrently!
```

Comment:
nit: it's not exported, but can we stick to godoc style for comments? (That is: `// scratchBuff is a reusable protobuf buffer. It is not safe to share between goroutines`).